storm-core/src/jvm/org/apache/storm/container/cgroup/core/MemoryCore.java

```diff
@@ -0,0 +1,189 @@
+        public final long cacheSize;
+        public final long rssSize;
+        public final long mappedFileSize;
+        public final long pgpginNum;
+        public final long pgpgoutNum;
+        public final long swapSize;
+        public final long activeAnonSize;
+        public final long inactiveAnonSize;
+        public final long activeFileSize;
+        public final long inactiveFileSize;
+        public final long unevictableSize;
+        public final long hierarchicalMemoryLimitSize;
+        public final long hierarchicalMemswLimitSize;
+        public final long totalCacheSize;
+        public final long totalRssSize;
+        public final long totalMappedFileSize;
+        public final long totalPgpginNum;
+        public final long totalPgpgoutNum;
+        public final long totalSwapSize;
+        public final long totalActiveAnonSize;
+        public final long totalInactiveAnonSize;
+        public final long totalActiveFileSize;
+        public final long totalInactiveFileSize;
+        public final long totalUnevictableSize;
+        public final long totalHierarchicalMemoryLimitSize;
+        public final long totalHierarchicalMemswLimitSize;
+
+        public Stat(String output) {
+            String[] splits = output.split("\n");
+            this.cacheSize = Long.parseLong(splits[0]);
+            this.rssSize = Long.parseLong(splits[1]);
+            this.mappedFileSize = Long.parseLong(splits[2]);
+            this.pgpginNum = Long.parseLong(splits[3]);
+            this.pgpgoutNum = Long.parseLong(splits[4]);
+            this.swapSize = Long.parseLong(splits[5]);
+            this.inactiveAnonSize = Long.parseLong(splits[6]);
+            this.activeAnonSize = Long.parseLong(splits[7]);
+            this.inactiveFileSize = Long.parseLong(splits[8]);
+            this.activeFileSize = Long.parseLong(splits[9]);
+            this.unevictableSize = Long.parseLong(splits[10]);
+            this.hierarchicalMemoryLimitSize = Long.parseLong(splits[11]);
+            this.hierarchicalMemswLimitSize = Long.parseLong(splits[12]);
+            this.totalCacheSize = Long.parseLong(splits[13]);
+            this.totalRssSize = Long.parseLong(splits[14]);
+            this.totalMappedFileSize = Long.parseLong(splits[15]);
+            this.totalPgpginNum = Long.parseLong(splits[16]);
+            this.totalPgpgoutNum = Long.parseLong(splits[17]);
+            this.totalSwapSize = Long.parseLong(splits[18]);
+            this.totalInactiveAnonSize = Long.parseLong(splits[19]);
+            this.totalActiveAnonSize = Long.parseLong(splits[20]);
+            this.totalInactiveFileSize = Long.parseLong(splits[21]);
+            this.totalActiveFileSize = Long.parseLong(splits[22]);
+            this.totalUnevictableSize = Long.parseLong(splits[23]);
+            this.totalHierarchicalMemoryLimitSize = Long.parseLong(splits[24]);
+            this.totalHierarchicalMemswLimitSize = Long.parseLong(splits[25]);
```

Comment:
Just a few minor comments, the core implementation of cgroup manager seems fine, not sure how other subsytems come into the picture if we are using only cpu and memory. @revans2 might want to take a look at it before it gets merged.